plugins/cloud-aws/src/test/java/org/elasticsearch/discovery/ec2/AmazonEC2Mock.java

```
@@ -0,0 +1,1358 @@
+                    }
+
+                    List<String> tags = expectedTags.get(tagName);
+                    if (tags == null) {
+                        tags = new ArrayList<>();
+                        expectedTags.put(tagName, tags);
+                    }
+                    tags.addAll(filter.getValues());
+                }
+            }
+
+            if (tagFiltered) {
+                logger.debug("--> expected tags: [{}]", expectedTags);
+                logger.debug("--> instance tags: [{}]", instanceTags);
+
+                instanceFound = true;
+                for (Map.Entry<String, List<String>> expectedTagsEntry : expectedTags.entrySet()) {
+                    List<String> instanceTagValues = instanceTags.get(expectedTagsEntry.getKey());
+                    if (instanceTagValues == null) {
+                        instanceFound = false;
+                        break;
+                    }
+
+                    for (String expectedValue : expectedTagsEntry.getValue()) {
+                        boolean valueFound = false;
+                        for (String instanceTagValue : instanceTagValues) {
+                            if (instanceTagValue.equals(expectedValue)) {
+                                valueFound = true;
+                            }
+                        }
+                        if (valueFound == false) {
+                            instanceFound = false;
+                        }
+                    }
+                }
+            }
+
+            if (tagFiltered == false || instanceFound) {
+                logger.debug("--> instance added");
+                filteredInstances.add(instance);
+            } else {
+                logger.debug("--> instance filtered");
+            }
+        }
+
+        return new DescribeInstancesResult().withReservations(
+                new Reservation().withInstances(filteredInstances)
+        );
+    }
+
+    // Not implemented methods in Mock
```

can we throw an UnsupportedOperationException in all those methods to make sure no once uses them?